edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -248,14 +247,6 @@ private async Task UpsertDeployments(V1ServiceList currentServices, V1Deployment
                     // Get the converted pod
                     V1PodTemplateSpec v1PodSpec = kubernetesModelBuilder.GetPod();
 
-                    // if this is the edge agent's deployment then it needs to run under a specific service account
```

Comment:
Don't we need this if we are launching EA?

Reply:
nope. iotedged provide the value for EA

Reply:
Wouldn't that have been through SA name? I am confused why this was needed before and no longer needed - can you elaborate on what changed?

Reply:
It was needed before because only EA had a service account with a special name. Now every module will have its own service account that equals sanitized module name. So no need to pass env variable for it anymore

Reply:
That makes sense!